src/ode_solver/rrk_explicit_ode_solver.cpp

```diff
@@ -0,0 +1,109 @@
+#include "rrk_explicit_ode_solver.h"
+
+namespace PHiLiP {
+namespace ODE {
+
+template <int dim, typename real, typename MeshType>
+RRKExplicitODESolver<dim,real,MeshType>::RRKExplicitODESolver(std::shared_ptr< DGBase<dim, real, MeshType> > dg_input)
+        : ExplicitODESolver<dim,real,MeshType>(dg_input)
+        {}
+
+template <int dim, typename real, typename MeshType>
+void RRKExplicitODESolver<dim,real,MeshType>::step_in_time (real dt, const bool pseudotime)
+{  
+
+    if (pseudotime){
+        this->pcout << "Pseudotime not implemented for RRK explicit. Aborting..." << std::endl;
+        std::abort();
+    }
+
+    this->solution_update = this->dg->solution; //storing u_n
+    
+    //calculating stages **Note that rk_stage[i] stores the RHS at a partial time-step (not solution u)
+    for (int i = 0; i < this->rk_order; ++i){
+
+        this->rk_stage[i]=0.0; //resets all entries to zero
+        
+        for (int j = 0; j < i; ++j){
+            if (this->butcher_tableau_a[i][j] != 0){
+                this->rk_stage[i].add(this->butcher_tableau_a[i][j], this->rk_stage[j]);
+            }
+        } //sum(a_ij *k_j)
+        
+        this->rk_stage[i]*=dt; 
+        
+        this->rk_stage[i].add(1.0,this->solution_update); //u_n + dt * sum(a_ij * k_j)
+
+        this->dg->solution = this->rk_stage[i];
+        this->dg->assemble_residual(); //RHS : du/dt = RHS = F(u_n + dt* sum(a_ij*k_j))
+        this->dg->global_inverse_mass_matrix.vmult(this->rk_stage[i], this->dg->right_hand_side); //rk_stage[i] = IMM*RHS = F(u_n + dt*sum(a_ij*k_j))
+    }
+
+    real relaxation_parameter = compute_relaxation_parameter_explicit();
+    //assemble solution from stages
+    for (int i = 0; i < this->rk_order; ++i){
+       this->solution_update.add(
+               dt * relaxation_parameter * this->butcher_tableau_b[i],
+               this->rk_stage[i]); 
+    }
+    this->dg->solution = this->solution_update; // u_np1 = u_n +gamma*dt* sum(k_i * b_i)
+
+    ++(this->current_iteration);
+    this->current_time += relaxation_parameter * dt;
```

Comment:
this seems to be the only difference between `ExplicitODESolver`'s `step_in_time()` function, consider defining a virtual function in `ExplicitODESolver` called `update_current_time` and by default would do `this->current_time += dt;` and in this class, it would do `real relaxation_parameter = compute_relaxation_parameter_explicit(); this->current_time += relaxation_parameter * dt;` (overriding the function)

Reply:
I was considering doing something like that but was having some difficulty with the best way to structure it. 

The "stage construction" loop (beginning at line 23) is identical to that in `explicit_ode_solver.cpp`. However, the "stage assembly" loop (beginning at line 44) is modified by the relaxation parameter. And, as you pointed out, the time increment is also modified.

That would mean that the options are either to repeat code (as I currently have done) or have `step_in_time()` look something like:
`construct_rk_stages()` -> loop beginning at line 23, same for explicit and RRK
`assemble_rk_stages()` -> loop beginning at line 44, RRK modifies by calculating `relaxation_parameter` and scaling the update.
`update_current_time()` -> as you pointed out; RRK modifies the time increment.

When first writing the code, I thought it would be most straightforward/readable to repeat `step_in_time()`, but I could also restructure as I outlined above. Do you think the restructuring I've proposed is a good idea?